sdk/v2/core/events.go

```diff
@@ -0,0 +1,361 @@
+	// ShortTitle is an optional, succinct title for the Event, ideal for use in
+	// lists or in scenarios where UI real estate is constrained.
+	ShortTitle string `json:"shortTitle,omitempty"`
+	// LongTitle is an optional, detailed title for the Event.
+	LongTitle string `json:"longTitle,omitempty"`
+	// Git contains git-specific Event details. These can be used to override
+	// similar details defined at the Project level. This is useful for scenarios
+	// wherein an Event may need to convey an alternative source, branch, etc.
+	Git *EventGitConfig `json:"git,omitempty"`
+	// Payload optionally contains Event details provided by the upstream system
+	// that was the original source of the event. Payloads MUST NOT contain
+	// sensitive information. Since Projects SUBSCRIBE to Events, the potential
+	// exists for any Project to express an interest in any or all Events. This
+	// being the case, sensitive details must never be present in payloads. The
+	// common workaround work this constraint (which is also a sensible practice
+	// to begin with) is that event payloads may contain REFERENCES to sensitive
+	// details that are useful only to properly configured Workers.
+	Payload string `json:"payload,omitempty"`
+	// Worker contains details of the worker that will/is/has handle(d) the Event.
+	Worker *Worker `json:"worker,omitempty"`
+}
+
+// MarshalJSON amends Event instances with type metadata so that clients do not
+// need to be concerned with the tedium of doing so.
+func (e Event) MarshalJSON() ([]byte, error) {
+	type Alias Event
+	return json.Marshal(
+		struct {
+			meta.TypeMeta `json:",inline"`
+			Alias         `json:",inline"`
+		}{
+			TypeMeta: meta.TypeMeta{
+				APIVersion: meta.APIVersion,
+				Kind:       "Event",
+			},
+			Alias: (Alias)(e),
+		},
+	)
+}
+
+// EventGitConfig represents git-specific Event details. These may override
+// similar details set at the Project level.
+type EventGitConfig struct {
+	// CloneURL specifies the location from where a source code repository may
+	// be cloned.
+	CloneURL string `json:"cloneURL,omitempty"`
+	// Commit specifies a commit (by sha) to be checked out.
+	Commit string `json:"commit,omitempty"`
+	// Ref specifies a tag or branch to be checked out. If left blank, this will
+	// default to "master" at runtime.
+	Ref string `json:"ref,omitempty"`
+}
+
+// CancelManyEventsResult represents a summary of a mass Event cancellation
+// operation.
+type CancelManyEventsResult struct {
+	// Count represents the number of Events canceled.
+	Count int64 `json:"count"`
+}
+
+// DeleteManyEventsResult represents a summary of a mass Event deletion
+// operation.
+type DeleteManyEventsResult struct {
+	// Count represents the number of Events deleted.
+	Count int64 `json:"count"`
+}
+
+// EventsClient is the specialized client for managing Events with the Brigade
+// API.
+type EventsClient interface {
+	// Create creates a new Event.
+	Create(context.Context, Event) (EventList, error)
```

Comment:
Yes. That should be spelled out more clearly.

The reason has to do with event subscriptions. Since one might submit a new event that isn't for a specific project, but is matched to _subscribed_ projects, more than one event might actually be created. This returns all the events that were created. This way if you're a gateway, for instance, you end up with a discrete list of events that you can then do whatever with-- perhaps monitor for completion in order to report results upstream.

Reply:
lmk if the recent commits add sufficient clarity.

Reply:
Yup, thank you for the updates.